force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -34,19 +34,90 @@ public with sharing class DummyJSONCallout {
+		
+		// Set the endpoint URL
+		String endpoint = 'https://dummyjson.com/users/' + dummyUserId;
+		request.setEndpoint(endpoint);
+		
+		// Set the HTTP method to GET
+		request.setMethod('GET');
+		
+		// Set the header to accept JSON response
+		request.setHeader('Content-Type', 'application/json');
+		
+		try {
+			// Send the request
+			HttpResponse response = http.send(request);
+			
+			// Check if the response was successful (status code 200)
+			if (response.getStatusCode() == 200) {
+				// Parse the JSON response
+				Map<String, Object> responseMap = (Map<String, Object>) JSON.deserializeUntyped(response.getBody());
+				
+				// Extract relevant data from the response
+				String firstName = (String) responseMap.get('firstName');
+				String lastName = (String) responseMap.get('lastName');
+				String email = (String) responseMap.get('email');
+				String phone = (String) responseMap.get('phone');
+				String birthDateStr = (String) responseMap.get('birthDate'); // assuming birthdate is a string in the format 'yyyy-MM-dd'
+				
+				// Convert birthDate to Date if it exists
+				Date birthDate = birthDateStr != null ? Date.valueOf(birthDateStr) : null;
+				
+				// Handle address extraction
+				Map<String, Object> addressMap = (Map<String, Object>) responseMap.get('address');
+				String street = addressMap != null && addressMap.containsKey('address') ? (String) addressMap.get('address') : null;
+				String city = addressMap != null && addressMap.containsKey('city') ? (String) addressMap.get('city') : null;
+				String postalCode = addressMap != null && addressMap.containsKey('postalCode') ? (String) addressMap.get('postalCode') : null;
+				String state = addressMap != null && addressMap.containsKey('state') ? (String) addressMap.get('state') : null;
+				String country = addressMap != null && addressMap.containsKey('country') ? (String) addressMap.get('country') : null;
+				
+				// Query for existing contact using the dummyUserId (DummyJSON_Id__c is an external id field)
+				Contact contact = [SELECT Id, DummyJSON_Id__c FROM Contact WHERE DummyJSON_Id__c = :dummyUserId LIMIT 1];
+				
+				// If contact exists, update the contact details, otherwise create a new one
+				if (contact != null) {
+					contact.FirstName = firstName;
+					contact.LastName = lastName;
+					contact.Email = email;
+					contact.Phone = phone;
+					contact.Birthdate = birthDate;
+					contact.MailingStreet = street;
+					contact.MailingCity = city;
+					contact.MailingPostalCode = postalCode;
+					contact.MailingState = state;
+					contact.MailingCountry = country;
+				} else {
+					contact = new Contact(
+						FirstName = firstName,
+						LastName = lastName,
+						Email = email,
+						Phone = phone,
+						Birthdate = birthDate,
+						MailingStreet = street,
+						MailingCity = city,
+						MailingPostalCode = postalCode,
+						MailingState = state,
+						MailingCountry = country,
+						DummyJSON_Id__c = dummyUserId
+					);
```

Comment:
Try to avoid duplicating like this. If contact is null, do only `contact = new Contact();` Then you can set all of the properties on contact, since they are same in both of your conditions here. Of course, the check is really not necessary at all, see the above comment.